Sources/Flatten.swift

```diff
@@ -916,10 +907,8 @@ extension Signal {
 			case let .value(value):
 				observer.send(value: value)
 			case let .failed(error):
-				handler(error).startWithSignal { signal, disposable in
-					serialDisposable.inner = disposable
-					signal.observe(observer)
-				}
+				serialDisposable.dispose()
```

Comment:
`serialDisposable` seems like it might be unnecessary? I don't see how it has any effect since the upstream doesn't need to be disposed when it has errored.